pkg/api/configsync/v1alpha1/reposync_types.go

```diff
@@ -41,7 +41,41 @@ type RepoSync struct {
 
 // RepoSyncSpec defines the desired state of a RepoSync.
 type RepoSyncSpec struct {
-	Spec `json:",inline"`
+	// sourceFormat specifies how the repository is formatted.
+	// See documentation for specifics of what these options do.
+	//
+	// Must be one of hierarchy, unstructured. Optional. Set to
+	// hierarchy if not specified.
+	//
+	// The validation of this is case-sensitive.
+	// +kubebuilder:validation:Pattern=^(hierarchy|unstructured|)$
```

Comment:
RepoSync only allows unstructured format.

Reply:
Sure, since for v1alpha1, it used to share the Spec between RootSync and RepoSync, v1alpha1.RepoSync used to allow hierarchy. FYI

Reply:
Done.

Reply:
ACK. Definitely something needs to be fixed.